eZ/Publish/Core/MVC/Symfony/Security/UserWrapped.php

```diff
@@ -45,20 +50,36 @@ public function __toString()
     public function setAPIUser(APIUser $apiUser)
     {
         $this->apiUser = $apiUser;
+        $this->apiUserReference = new UserReference($apiUser->getUserId());
     }
 
     /**
      * @return \eZ\Publish\API\Repository\Values\User\User
      */
     public function getAPIUser()
     {
+        if (!$this->apiUser instanceof APIUser) {
+            throw new \LogicException(
+                'Attempted to get APIUser before it has been set by UserProvider, APIUser is not serialized to session'
+            );
+        }
```

Comment:
The jira issue has an example.  You can see a simple gist here that reproduces the error

https://gist.github.com/wizhippo/06f480a44ff8d114852385b478eee726

Reply:
> 3. Please add the test coverage for the reported use case.

Ideas how would be helpful.  The issue is that the apiUser gets serialized to the session and as it is a proxy when it's location gets serialized the location's content gets serialized and in turn the content gets serialized and the loop continues until a crash.

I believe besides just having a smaller session the standard User serializes to a UserReference to avoid this.  A Wrapped user should so the same.

The test for refreshing the user from a UserReference already exists and passes.

Reply:
@alongosz See https://github.com/ezsystems/ezplatform-kernel/pull/70#issuecomment-646032047 regarding testing. Let me know if the changes are acceptable.